### PR TITLE
Add ssh slaves to CI

### PR DESCRIPTION
**Pull request by surminus**
See commits for detail. This is a PR to replace the current Jenkins swarm plugin with the SSH slave plugin instead as we are seeing stability issues with the swarm protocol.

### CODE REVIEW

File: modules/govuk_ci/manifests/master.pp [hunk 1]
```diff
@@ -57,4 +59,9 @@
     mode => '0555',
   }
 
+  create_resources('::govuk_jenkins::ssh_slave', $ci_agents, {
+    'credentials_id' => $credentials_id,
+    'labels'         => 'mongo-3.2',
```
Comment (deanwilson) [inline modules/govuk_ci/manifests/master.pp]: Is this to ensure that all CI hosts have this version of mongo? Should this be in the hiera with the other labels?

File: modules/govuk_jenkins/manifests/ssh_slave.pp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+#
+# [*jenkins_user*]
+#   The user which the master connects to the agent over SSH as.
+#
+define govuk_jenkins::ssh_slave (
+  $agent_hostname,
+  $credentials_id,
+  $labels = undef,
+  $description = 'Generic build agent',
+  $executors = '4',
```
Comment (surminus) [inline modules/govuk_jenkins/manifests/ssh_slave.pp]: I did it to annoy you ;) I'll change them. There's no reason other than absent mindedness.

File: modules/govuk_ci/manifests/master.pp (hunk 1 above)
Comment (surminus) [inline modules/govuk_ci/manifests/master.pp]: This is a default label. I put it in here to try and cut down the default hiera, but it probably makes more sense to be verbose in the hiera. WIll update.

File: hieradata/common.yaml [hunk 3]
```diff
@@ -635,6 +635,26 @@ govuk_ci::master::pipeline_jobs:
   ubuntu_unused_kernels:
     repo_owner: 'gds-operations'
 
+govuk_ci::master::ci_agents:
+  ci-agent-1:
+    agent_hostname: 'ci-agent-1.ci'
+    description: 'Build agent with docker'
+    labels: 'mongodb-2.4 docker'
```
Comment (djsd123) [inline hieradata/common.yaml]: Would you please do me a solid and amend the labels so they match the labels in our Jenkinsfiles?

i.e. `mongo-3.2` becomes `mongodb-3.2`

Comment (surminus) [inline hieradata/common.yaml]: @djsd123 sorry yeah, I spotted that.

**Comment by surminus** [comment]
I've deployed this to Integration (and stopped puppet on the master and agents). I believe it works as intended, and after some testing I've fixed some bugs.

**Pull request merged**